pkg/apis/pubsub/v1alpha1/pull_subscription_types.go

```diff
@@ -89,27 +88,51 @@ type PullSubscriptionSpec struct {
 	// RetentionDuration defines how long to retain messages in backlog, from
 	// the time of publish. If RetainAckedMessages is true, this duration
 	// affects the retention of acknowledged messages, otherwise only
-	// unacknowledged messages are retained. Defaults to 7 days. Cannot be
-	// longer than 7 days or shorter than 10 minutes.
+	// unacknowledged messages are retained. Cannot be longer than 7 days or
+	// shorter than 10 minutes. Defaults to 7 days ('7d').
 	// +optional
-	RetentionDuration *time.Duration `json:"retentionDuration,omitempty"`
+	RetentionDuration *string `json:"retentionDuration,omitempty"`
 
-	// Sink is a reference to an object that will resolve to a domain name to
-	// use as the sink.
-	// +optional
-	Sink *corev1.ObjectReference `json:"sink,omitempty"`
+	// Sink is a reference to an object that will resolve to a domain name or a
+	// URI directly to use as the sink.
+	Sink *Destination `json:"sink,omitempty"`
 
 	// Transformer is a reference to an object that will resolve to a domain
-	// name to use as the transformer.
+	// name or a URI directly to use as the transformer or a URI directly.
 	// +optional
-	Transformer *corev1.ObjectReference `json:"transformer,omitempty"`
+	Transformer *Destination `json:"transformer,omitempty"`
```

Comment:
this makes sense to be a pointer.

Reply:
Now sink is a pointer.

Reply:
it was supposed to be paired with the comment about Sink not being a pointer, but this makes sense being a pointer. The comments were supposed to be linked but whatevs. I was trying to make sure you don't change them both to a non-pointer :)